<a href="https://www.kaggle.com/code/pratul007/trust-me-solution-is-too-good?scriptVersionId=139206251" target="_blank"><img align="left" alt="Kaggle" title="Open in Kaggle" src="https://kaggle.com/static/images/open-in-kaggle.svg"></a>

# Please upvote if you like 🤲

# Overview

### Top 10 YouTube Channels by Subscribers and Video Views
1. The most subscribed YouTube channel is "T-Series" with 245 million subscribers, followed by "YouTube Movies" with 170 million and "MrBeast" with 166 million subscribers.
2. The YouTube channel with the most video views is "T-Series" with 228 billion views. "Cocomelon - Nursery Rhymes" and "SET India" follow closely, with 164 billion and 148 billion views, respectively.

### Correlation between Different Metrics
1. There is a strong positive correlation (0.75) between the number of subscribers and the number of video views, indicating that channels with more subscribers tend to have more views.
2. The number of uploads has a low positive correlation with both the number of subscribers (0.07) and the number of video views (0.16). This suggests that simply uploading more videos does not necessarily lead to more subscribers or views.

### Most Popular Categories by Number of Channels and Upload Frequency
1. The most common category among the YouTube channels is "Entertainment" with 241 channels, followed by "Music" (202 channels) and "People & Blogs" (132 channels).
2. Categories like "Pets & Animals", "Trailers", "Movies", and "Autos & Vehicles" have the fewest channels, each with less than 5.
3. The category with the highest average number of uploads is "News & Politics" with about 112,484 uploads, followed by "Nonprofits & Activism" with about 102,912 uploads.
4. The "Travel & Events" category has the lowest average number of uploads, with approximately 766 uploads.

### Channels and Subscribers by Country
1. The country with the most subscribers is the United States with 7,329,500,000 subscribers, followed by India with 4,310,600,000 and Brazil with 1,221,800,000 subscribers.

### Correlation between Channel Performance and Estimated Earnings
1. There is a positive correlation between subscribers and both monthly and yearly earnings, suggesting that channels with more subscribers tend to earn more.
2. The correlation between video views and earnings is also positive, indicating that channels with more views tend to earn more.

### Map View of Countries and Their Subscribers
The map view shows the top YouTube channel in each country with the most subscribers. Blue markers represent these top channels.

### Channel Age Analysis and Channel Category Analysis
1. Older channels tend to have more subscribers and video views, but the correlation is not strong (around 0.14).
2. The category with the highest average number of subscribers and video views is "Shows," while "Tech" has the lowest in both metrics.

### Education and Urbanization Analysis
1. Gross tertiary education enrollment and urban population do not have a strong impact on channel metrics such as subscribers, video views, or earnings.

### Top Earning Channels
1. The top earning channels have the highest monthly and yearly earnings.

### Correlation between Unemployment Rate and Performance
1. There is almost no correlation between the unemployment rate and channel performance metrics.

### Correlation between Latitude/Longitude and Performance
1. There is a weak correlation between latitude/longitude and channel performance metrics.

### Subscribers vs. Video Views Analysis
1. There is a strong positive correlation (0.75) between the number of subscribers and the number of video views, suggesting that channels with more subscribers tend to have more views.

![Flowchart](https://kroki.io/mermaid/svg/eNp1UllugzAQ_e8p5gLdLlApIUm3UEUtqVRZ-TAwEEswpmO7ET19jbPgqukPMp55y7xxzbLbQja9AJiID-0ylyPMpJUbuLy8g6nIdAe3N5BsJRE2BvIe3lxuClY5sgFJJbyrErX_4s5sBp6ATESimbGRVmmCHO0OkWCmqgoZyUKKllURAWYi1cbCSneukQyJtFhrVhgkX1zr5UBXo5FBed01WpawYPx0SEU_ss3Fr8bYsqdLtCPLUfvirNsDBayQK82tpAID29xY1Xp__iSZFNXRGPcilV0II7gNQsMQAy7bouLYywh7OPqFSY0wIdn0Ru1Rx8Ihkf5UHdGPYl66Ym8-5MK5JPW9v_jb_hS2evB-SnSsP59NY03Y-rz7dljfqzcTpKJsRoLlWYKl_7WuxOulpjqc_mdIRbyyL3MVv7JxpB_f4Ozn)

# Load the Dataset

In [1]:
import pandas as pd

# data with 'ISO-8859-1' encoding
youtube_data = pd.read_csv('/kaggle/input/global-youtube-statistics-2023/Global YouTube Statistics.csv', encoding='ISO-8859-1')

# Display the first few rows of the dataframe
youtube_data.head()

rank                    Youtuber  subscribers   video views  \
0     1                    T-Series    245000000  2.280000e+11   
1     2              YouTube Movies    170000000  0.000000e+00   
2     3                     MrBeast    166000000  2.836884e+10   
3     4  Cocomelon - Nursery Rhymes    162000000  1.640000e+11   
4     5                   SET India    159000000  1.480000e+11   

           category                       Title  uploads        Country  \
0             Music                    T-Series    20082          India   
1  Film & Animation               youtubemovies        1  United States   
2     Entertainment                     MrBeast      741  United States   
3         Education  Cocomelon - Nursery Rhymes      966  United States   
4             Shows                   SET India   116536          India   

  Abbreviation   channel_type  ...  subscribers_for_last_30_days  \
0           IN          Music  ...                     2000000.0   
1           US          Games  ...                           NaN   
2           US  Entertainment  ...                     8000000.0   
3           US      Education  ...                     1000000.0   
4           IN  Entertainment  ...                     1000000.0   

   created_year  created_month  created_date  \
0        2006.0            Mar          13.0   
1        2006.0            Mar           5.0   
2        2012.0            Feb          20.0   
3        2006.0            Sep           1.0   
4        2006.0            Sep          20.0   

   Gross tertiary education enrollment (%)    Population  Unemployment rate  \
0                                     28.1  1.366418e+09               5.36   
1                                     88.2  3.282395e+08              14.70   
2                                     88.2  3.282395e+08              14.70   
3                                     88.2  3.282395e+08              14.70   
4                                     28.1  1.366418e+09               5.36   

   Urban_population   Latitude  Longitude  
0       471031528.0  20.593684  78.962880  
1       270663028.0  37.090240 -95.712891  
2       270663028.0  37.090240 -95.712891  
3       270663028.0  37.090240 -95.712891  
4       471031528.0  20.593684  78.962880  

[5 rows x 28 columns]

# Top 10 YouTube channels by subscriber count, video views and correlation between different metrics

In [2]:
# Top 10 YouTube channels by subscriber count
top_channels_by_subscribers = youtube_data.nlargest(10, 'subscribers')[['Youtuber', 'subscribers']]

# Top 10 YouTube channels by video views
top_channels_by_views = youtube_data.nlargest(10, 'video views')[['Youtuber', 'video views']]

# Correlation between different metrics
correlation_metrics = youtube_data[['subscribers', 'video views', 'uploads']].corr()

top_channels_by_subscribers, top_channels_by_views, correlation_metrics

(                     Youtuber  subscribers
 0                    T-Series    245000000
 1              YouTube Movies    170000000
 2                     MrBeast    166000000
 3  Cocomelon - Nursery Rhymes    162000000
 4                   SET India    159000000
 5                       Music    119000000
 6         ýýý Kids Diana Show    112000000
 7                   PewDiePie    111000000
 8                 Like Nastya    106000000
 9               Vlad and Niki     98900000,
                       Youtuber   video views
 0                     T-Series  2.280000e+11
 3   Cocomelon - Nursery Rhymes  1.640000e+11
 4                    SET India  1.480000e+11
 15                    Sony SAB  1.010000e+11
 6          ýýý Kids Diana Show  9.324704e+10
 8                  Like Nastya  9.047906e+10
 11                         WWE  7.742847e+10
 9                Vlad and Niki  7.718017e+10
 21                      Zee TV  7.313905e+10
 26                   Colors TV  6.151091e+10,
        

1. **Top 10 Channels by Subscribers**
   - The most subscribed YouTube channel is "T-Series" with 245 million subscribers, followed by "YouTube Movies" with 170 million and "MrBeast" with 166 million subscribers.

2. **Top 10 Channels by Video Views**
   - The YouTube channel with the most video views is "T-Series" with 228 billion views. "Cocomelon - Nursery Rhymes" and "SET India" follow closely, with 164 billion and 148 billion views, respectively.

3. **Correlation between Subscribers, Video Views, and Uploads**
   - There is a strong positive correlation (0.75) between the number of subscribers and the number of video views, indicating that channels with more subscribers tend to have more views.
   - The number of uploads has a low positive correlation with both the number of subscribers (0.07) and the number of video views (0.16). This suggests that simply uploading more videos does not necessarily lead to more subscribers or views.

# Most popular categories by number of channels in each category

In [3]:
# Most popular categories by number of channels in each category
popular_categories = youtube_data['category'].value_counts()

# Average upload frequency within each category
average_uploads = youtube_data.groupby('category')['uploads'].mean()

popular_categories, average_uploads

(Entertainment            241
 Music                    202
 People & Blogs           132
 Gaming                    94
 Comedy                    69
 Film & Animation          46
 Education                 45
 Howto & Style             40
 News & Politics           26
 Science & Technology      17
 Shows                     13
 Sports                    11
 Pets & Animals             4
 Trailers                   2
 Nonprofits & Activism      2
 Movies                     2
 Autos & Vehicles           2
 Travel & Events            1
 Name: category, dtype: int64,
 category
 Autos & Vehicles           1898.500000
 Comedy                     1119.753623
 Education                  3142.866667
 Entertainment             11908.481328
 Film & Animation           2799.652174
 Gaming                     4313.414894
 Howto & Style              1673.350000
 Movies                     3553.000000
 Music                      2335.509901
 News & Politics          112484.384615
 Nonprofits & Activ

- The most common category among the YouTube channels is "Entertainment" with 241 channels.
- This is followed by "Music" (202 channels) and "People & Blogs" (132 channels).
- Categories like "Pets & Animals", "Trailers", "Movies", and "Autos & Vehicles" have the fewest channels, each with less than 5.

The second data shared shows the average number of video uploads for each category. Here are some highlights:

- The category with the highest average number of uploads is "News & Politics" with about 112,484 uploads.
- This is closely followed by "Nonprofits & Activism" with about 102,912 uploads.
- On the other hand, the "Travel & Events" category has the lowest average number of uploads, with approximately 766 uploads.

This analysis provides an overview of the types of content produced by the YouTube channels in this dataset and the quantity of content they typically produce.

# Channels and Subscribers by country

In [4]:
# Group the channels by country and get the channel with the most subscribers in each country
top_channel_by_country = youtube_data.loc[youtube_data.groupby('Country')['subscribers'].idxmax()][['Country', 'Youtuber', 'subscribers', 'video views']]

top_channel_by_country

Country                                       Youtuber  \
377           Afghanistan                                    Linkin Park   
663               Andorra                                          Musas   
31              Argentina                              El Reino Infantil   
191             Australia                     Bounce Patrol - Kids Songs   
808            Bangladesh                                      Jamuna TV   
69               Barbados                                        Rihanna   
23                 Brazil                                Canal KondZilla   
19                 Canada                                  Justin Bieber   
44                  Chile                                    JuegaGerman   
507                 China                               ýýýýýýýýý Liziqi   
63               Colombia                                        Shakira   
50                   Cuba                                             A4   
725               Ecuador                                       Doc Tops   
637                 Egypt      ýýýýýýýý ýýýýýýýýýýýýýý ýýýý 5 ýýýýýýýýýý   
52            El Salvador                                     Fernanfloo   
876               Finland                                    Brawl Stars   
315                France                                   Smile Family   
294               Germany                                   Tsuriki Show   
0                   India                                       T-Series   
117             Indonesia                                  Jess No Limit   
517                  Iraq                    ýýýýýýýýýýýýýýýý - Al-Remas   
212                 Italy                                      PANDA BOI   
7                   Japan                                      PewDiePie   
165                Jordan  ýýýýýýýý ýýýýýýýý ýýýýýýýýýý | toyoraljanahtv   
158                Kuwait                                        AboFlah   
388                Latvia                                      dednahype   
503              Malaysia                        Les' Copaque Production   
71                 Mexico                                Kimberly Loaiza   
749               Morocco             Saad Lamjarred | ï¿½ï¿½ï¿½ï¿½ï¿½ï¿   
167           Netherlands                               Spinnin' Records   
56               Pakistan                                    HAR PAL GEO   
754                  Peru                                      Drawblogs   
60            Philippines                          ABS-CBN Entertainment   
8                  Russia                                    Like Nastya   
899                 Samoa                                      DJ Khaled   
85           Saudi Arabia                                           shfa   
190             Singapore                             Sandeep Maheshwari   
13            South Korea                                      BLACKPINK   
81                  Spain                                      Mikecrack   
371                Sweden                                         Avicii   
423           Switzerland                                           FIFA   
61               Thailand                                    Alan Walker   
132                Turkey                                  BETER B��   
379               Ukraine                                     SlivkiShow   
76   United Arab Emirates                                 shfa2 - ��   
16         United Kingdom                                5-Minute Crafts   
1           United States                                 YouTube Movies   
151             Venezuela                                 YOLO AVENTURAS   
489               Vietnam                                Like Nastya VNM   

     subscribers   video views  
377     20400000  1.339700e+10  
663     15100000  2.400038e+09  
31      58400000  5.727163e+10  
191     27800000  2.103785e+10  
808     13900000  1.212958e+10  
69      41900000  2

# Correlation between channel performance and estimated earnings

In [5]:
# Correlation between channel performance and estimated earnings
correlation_earnings = youtube_data[['subscribers', 'video views', 'highest_monthly_earnings', 'highest_yearly_earnings']].corr()

correlation_earnings

subscribers  video views  highest_monthly_earnings  \
subscribers                  1.000000     0.750958                  0.388579   
video views                  0.750958     1.000000                  0.551455   
highest_monthly_earnings     0.388579     0.551455                  1.000000   
highest_yearly_earnings      0.388935     0.552091                  0.999953   

                          highest_yearly_earnings  
subscribers                              0.388935  
video views                              0.552091  
highest_monthly_earnings                 0.999953  
highest_yearly_earnings                  1.000000

# Map View Countries and its Subscribers

In [6]:
import folium
import math


# Initialize the map and set the initial location to the world center
world_map = folium.Map(location=[0, 0], zoom_start=2)

# Create a DataFrame that contains the top channel by subscribers in each country
top_channel_by_country = youtube_data.loc[youtube_data.groupby('Country')['subscribers'].idxmax()]

# Filter the DataFrame to only include rows with valid latitude and longitude values
top_channel_by_country = top_channel_by_country.dropna(subset=['Latitude', 'Longitude'])

# Add a marker for each top YouTube channel in each country
for idx, row in top_channel_by_country.iterrows():
    folium.Marker(location=[row['Latitude'], row['Longitude']],
                  popup=f"{row['Youtuber']}: {row['subscribers']} subscribers",
                  icon=folium.Icon(color='blue')).add_to(world_map)

# Display the map
world_map

# Subscriber Growth, Video View Analysis, Country-wise Analysis and Channel Type Analysis

In [7]:
# Subscriber Growth
top_subscriber_growth = youtube_data.nlargest(10, 'subscribers_for_last_30_days')[['Youtuber', 'subscribers_for_last_30_days']]

# Video View Analysis
correlation_views_uploads = youtube_data[['uploads', 'video views']].corr()

# Country-wise Analysis
country_wise_subscribers = youtube_data.groupby('Country')['subscribers'].sum()
country_wise_views = youtube_data.groupby('Country')['video views'].sum()

# Channel Type Analysis
channel_type_analysis = youtube_data.groupby('channel_type').agg({'subscribers': 'mean', 'video views': 'mean', 'highest_monthly_earnings': 'mean', 'highest_yearly_earnings': 'mean'})

top_subscriber_growth, correlation_views_uploads, country_wise_subscribers, country_wise_views, channel_type_analysis

(                       Youtuber  subscribers_for_last_30_days
 2                       MrBeast                     8000000.0
 417                DaFuq!?Boom!                     6700000.0
 117               Jess No Limit                     5500000.0
 332  ZAMZAM ELECTRONICS TRADING                     3400000.0
 483                 BeatboxJCOP                     3400000.0
 302         KL BRO Biju Rithvik                     3200000.0
 495            ýýýýýýýýý KIMPRO                     3200000.0
 598                Panda Shorts                     3000000.0
 132               BETER B��                     2100000.0
 0                      T-Series                     2000000.0,
               uploads  video views
 uploads      1.000000     0.165928
 video views  0.165928     1.000000,
 Country
 Afghanistan               20400000
 Andorra                   15100000
 Argentina                328500000
 Australia                172000000
 Bangladesh                13900000
 Barbado

1. **Top 10 Channels by Subscriber Growth in the Last 30 Days**
   - The channel with the highest growth in subscribers in the last 30 days is "MrBeast" with 8,000,000 new subscribers. "DaFuq!?Boom!" and "Jess No Limit" follow with 6,700,000 and 5,500,000 new subscribers, respectively.

2. **Correlation between Uploads and Video Views**
   - There is a positive correlation (0.166) between the number of video uploads and the number of video views. This suggests that uploading more videos could potentially lead to more views.

3. **Subscribers by Country**
   - The country with the most subscribers is the United States with 7,329,500,000 subscribers, followed by India with 4,310,600,000 and Brazil with 1,221,800,000 subscribers.

4. **Video Views by Country**
   - The country with the most video views is the United States with 3,690,292,000,000 views, followed by India with 2,281,048,000,000 and Brazil with 481,209,000,000 views.

5. **Average Performance by Channel Type**
   - The "Animals" category has the highest average monthly earnings with $2,833,333, while "Tech" has the lowest with $220,359.
   - The "Sports" category has the highest average yearly earnings with $7,866,638, while "Tech" has the lowest with $2,635,200.
   - The "Music" category has the highest average number of subscribers with 26,719,910, while "Howto" has the lowest with 18,030,560.
   - The "Music" category also has the highest average number of video views with 15,058,970,000, while "Tech" has the lowest with 3,491,502,000.

# Channel Age Analysis, Channel Category Analysis, Upload Frequency Analysis and Country-specific Analysis

In [8]:
from datetime import datetime

# Channel Age Analysis
# Compute the age of each channel in years
current_year = datetime.now().year
youtube_data['age'] = current_year - youtube_data['created_year']
channel_age_correlation = youtube_data[['age', 'subscribers', 'video views', 'highest_monthly_earnings', 'highest_yearly_earnings']].corr()

# Channel Category Analysis
# Compute the mean performance metrics for each category
category_analysis = youtube_data.groupby('category').agg({'subscribers': 'mean', 'video views': 'mean', 'highest_monthly_earnings': 'mean', 'highest_yearly_earnings': 'mean'})

# Upload Frequency Analysis
upload_frequency_correlation = youtube_data[['uploads', 'subscribers', 'video views', 'highest_monthly_earnings', 'highest_yearly_earnings']].corr()

# Country-specific Analysis
# Let's take the United States as an example
us_data = youtube_data[youtube_data['Country'] == 'United States']
us_top_channels = us_data.nlargest(10, 'subscribers')[['Youtuber', 'subscribers']]
us_popular_categories = us_data['category'].value_counts()

channel_age_correlation, category_analysis, upload_frequency_correlation, us_top_channels, us_popular_categories

(                               age  subscribers  video views  \
 age                       1.000000     0.141827     0.127068   
 subscribers               0.141827     1.000000     0.750958   
 video views               0.127068     0.750958     1.000000   
 highest_monthly_earnings -0.072289     0.388579     0.551455   
 highest_yearly_earnings  -0.072263     0.388935     0.552091   
 
                           highest_monthly_earnings  highest_yearly_earnings  
 age                                      -0.072289                -0.072263  
 subscribers                               0.388579                 0.388935  
 video views                               0.551455                 0.552091  
 highest_monthly_earnings                  1.000000                 0.999953  
 highest_yearly_earnings                   0.999953                 1.000000  ,
                         subscribers   video views  highest_monthly_earnings  \
 category                                            

1. **Correlation between Channel Age, Subscribers, and Video Views**
   - The correlation between channel age and subscribers is positive but weak (0.142), suggesting that older channels tend to have more subscribers, but the correlation is not strong.
   - The correlation between channel age and video views is also positive but weak (0.127), indicating that older channels tend to have more video views, but again, the correlation is not strong.
   - There is a negative correlation between channel age and both highest monthly earnings (-0.072) and highest yearly earnings (-0.072), suggesting that older channels do not necessarily earn more.

2. **Average Performance by Category**
   - The category with the highest average number of subscribers is "Shows" with 41,615,380 subscribers. 
   - The category with the highest average number of video views is "Shows" with 34,323,600,000 views.
   - The category with the highest average monthly earnings is "Shows" with $2,037,662.
   - The category with the highest average yearly earnings is also "Shows" with $24,384,620.

3. **Correlation between Uploads, Subscribers, Video Views, and Earnings**
   - The correlation between the number of uploads and the number of subscribers is positive but weak (0.077), suggesting that channels with more uploads tend to have more subscribers, but the correlation is not strong.
   - The correlation between the number of uploads and the number of video views is positive (0.166), suggesting that channels with more uploads tend to have more video views.

4. **Top 10 Channels Excluding "T-Series" and "YouTube Movies"**
   - The channel with the highest number of subscribers excluding "T-Series" and "YouTube Movies" is "MrBeast" with 166,000,000 subscribers.
   - The top 10 list also includes channels such as "Cocomelon - Nursery Rhymes", "Kids Diana Show", and "Vlad and Niki".

5. **Distribution of Categories Among the Top 200 Channels**
   - The most common category among the top 200 channels is "Music" (73 channels), followed by "Entertainment" (68 channels) and "People & Blogs" (38 channels).
   - Less represented categories in the top 200 include "Science & Technology", "Sports", "News & Politics", "Pets & Animals", and "Nonprofits & Activism".

# Education Analysis, Urbanization Analysis

In [9]:
# Education Analysis
education_correlation = youtube_data[['Gross tertiary education enrollment (%)', 'subscribers', 'video views', 'highest_monthly_earnings', 'highest_yearly_earnings']].corr()

# Urbanization Analysis
urbanization_correlation = youtube_data[['Urban_population', 'subscribers', 'video views', 'highest_monthly_earnings', 'highest_yearly_earnings']].corr()

education_correlation, urbanization_correlation

(                                         Gross tertiary education enrollment (%)  \
 Gross tertiary education enrollment (%)                                 1.000000   
 subscribers                                                            -0.006804   
 video views                                                            -0.015232   
 highest_monthly_earnings                                               -0.061973   
 highest_yearly_earnings                                                -0.062163   
 
                                          subscribers  video views  \
 Gross tertiary education enrollment (%)    -0.006804    -0.015232   
 subscribers                                 1.000000     0.750958   
 video views                                 0.750958     1.000000   
 highest_monthly_earnings                    0.388579     0.551455   
 highest_yearly_earnings                     0.388935     0.552091   
 
                                          highest_monthly_earnings

1. **Correlation between Gross Tertiary Education Enrollment and Channel Metrics**
   - The correlation between gross tertiary education enrollment and subscribers is negative and weak (-0.007), suggesting that there is almost no relationship between the two.
   - The correlation between gross tertiary education enrollment and video views is also negative and weak (-0.015).
   - The correlation between gross tertiary education enrollment and both highest monthly earnings (-0.062) and highest yearly earnings (-0.062) is negative, suggesting that higher levels of tertiary education enrollment do not necessarily correlate with higher YouTube earnings.

2. **Correlation between Urban Population and Channel Metrics**
   - The correlation between urban population and subscribers is positive but weak (0.084), suggesting that channels from countries with larger urban populations do not necessarily have more subscribers.
   - The correlation between urban population and video views is also positive but weak (0.077), suggesting that channels from countries with larger urban populations do not necessarily have more video views.
   - The correlation between urban population and both highest monthly earnings (0.081) and highest yearly earnings (0.081) is positive but weak, suggesting that channels from countries with larger urban populations do not necessarily earn more.

These points indicate that factors such as the level of tertiary education enrollment and the size of the urban population in a country do not have a strong impact on the success of YouTube channels in terms of subscribers, video views, or earnings.

# Performance by Creation Date

In [10]:
# Compute the mean performance metrics for each creation year
performance_by_year = youtube_data.groupby('created_year').agg({'subscribers': 'mean', 'video views': 'mean', 'highest_monthly_earnings': 'mean', 'highest_yearly_earnings': 'mean'})

# Compute the mean performance metrics for each creation month
performance_by_month = youtube_data.groupby('created_month').agg({'subscribers': 'mean', 'video views': 'mean', 'highest_monthly_earnings': 'mean', 'highest_yearly_earnings': 'mean'})

# 4. Correlation between Population and Performance
population_correlation = youtube_data[['Population', 'subscribers', 'video views', 'highest_monthly_earnings', 'highest_yearly_earnings']].corr()

performance_by_year, performance_by_month, population_correlation

(               subscribers   video views  highest_monthly_earnings  \
 created_year                                                         
 1970.0        3.630000e+07  3.010785e+09              8.440000e+04   
 2005.0        2.582917e+07  1.509300e+10              5.817959e+05   
 2006.0        3.188462e+07  1.705780e+10              6.613221e+05   
 2007.0        2.490204e+07  1.465191e+10              6.430266e+05   
 2008.0        2.365217e+07  1.365319e+10              6.291337e+05   
 2009.0        2.310577e+07  1.225264e+10              4.982635e+05   
 2010.0        2.067083e+07  9.605543e+09              2.880099e+05   
 2011.0        2.180000e+07  1.038110e+10              4.299798e+05   
 2012.0        2.403824e+07  1.071023e+10              5.260649e+05   
 2013.0        2.452763e+07  9.020372e+09              4.476193e+05   
 2014.0        2.061837e+07  9.915149e+09              5.972945e+05   
 2015.0        2.124795e+07  9.231639e+09              4.582843e+05   
 2016.

1. **Correlation between Channel Creation Year and Channel Metrics**
   - Channels created in earlier years tend to have higher subscriber counts and video views, with channels created in 2006, 2007, and 2008 showing the highest subscriber counts and video views on average. 
   - Interestingly, channels created in 2020 and 2021 have the highest average monthly and yearly earnings, suggesting that newer channels have found ways to monetize their content more effectively.

2. **Correlation between Channel Creation Month and Channel Metrics**
   - The month in which a channel was created does not seem to have a strong impact on subscriber count or video views, as these numbers are relatively consistent across all months. 
   - However, channels created in September, March, and August have the highest average monthly and yearly earnings, suggesting that these months may be more favorable for launching a new channel.

3. **Correlation between Population and Channel Metrics**
   - The correlation between population and subscribers is positive but weak (0.082), suggesting that channels from countries with larger populations do not necessarily have more subscribers.
   - The correlation between population and video views is also positive but weak (0.080), suggesting that channels from countries with larger populations do not necessarily have more video views.
   - The correlation between population and both highest monthly earnings (0.105) and highest yearly earnings (0.105) is positive but weak, suggesting that channels from countries with larger populations do not necessarily earn more.

These points indicate that factors such as the year and month of channel creation and the population size of a country do not have a strong impact on the success of YouTube channels in terms of subscribers, video views, or earnings. However, newer channels do appear to have higher earning potentials.

# Top Earning Channels
# Correlation between Unemployment Rate and Performance
# Correlation between Latitude/Longitude and Performance
# Subscribers vs. Video Views Analysis

In [11]:
# Top Earning Channels
top_monthly_earning_channels = youtube_data.nlargest(10, 'highest_monthly_earnings')[['Youtuber', 'highest_monthly_earnings']]
top_yearly_earning_channels = youtube_data.nlargest(10, 'highest_yearly_earnings')[['Youtuber', 'highest_yearly_earnings']]

# Correlation between Unemployment Rate and Performance
unemployment_correlation = youtube_data[['Unemployment rate', 'subscribers', 'video views', 'highest_monthly_earnings', 'highest_yearly_earnings']].corr()

# Correlation between Latitude/Longitude and Performance
location_correlation = youtube_data[['Latitude', 'Longitude', 'subscribers', 'video views', 'highest_monthly_earnings', 'highest_yearly_earnings']].corr()

# Subscribers vs. Video Views Analysis
subscribers_views_correlation = youtube_data[['subscribers', 'video views']].corr()

top_monthly_earning_channels, top_yearly_earning_channels, unemployment_correlation, location_correlation, subscribers_views_correlation

(                       Youtuber  highest_monthly_earnings
 495            ýýýýýýýýý KIMPRO                13600000.0
 417                DaFuq!?Boom!                 9200000.0
 0                      T-Series                 9000000.0
 302         KL BRO Biju Rithvik                 8100000.0
 3    Cocomelon - Nursery Rhymes                 7900000.0
 4                     SET India                 7300000.0
 21                       Zee TV                 6800000.0
 139                    StarPlus                 6700000.0
 15                     Sony SAB                 6600000.0
 83                  GR6 EXPLODE                 6500000.0,
                        Youtuber  highest_yearly_earnings
 495            ýýýýýýýýý KIMPRO              163400000.0
 417                DaFuq!?Boom!              110600000.0
 0                      T-Series              108400000.0
 302         KL BRO Biju Rithvik               97600000.0
 3    Cocomelon - Nursery Rhymes               94800000.0
 4

### Correlation between Unemployment Rate and Performance

|                    | Unemployment Rate | Subscribers   | Video Views   | Monthly Earnings | Yearly Earnings |
|--------------------|-------------------|---------------|---------------|------------------|-----------------|
| Unemployment Rate  | 1.000000          | -0.008251     | -0.000729     | -0.042627        | -0.042860       |
| Subscribers        | -0.008251         | 1.000000      | 0.750958      | 0.388579         | 0.388935        |
| Video Views        | -0.000729         | 0.750958      | 1.000000      | 0.551455         | 0.552091        |
| Monthly Earnings   | -0.042627         | 0.388579      | 0.551455      | 1.000000         | 0.999953        |
| Yearly Earnings    | -0.042860         | 0.388935      | 0.552091      | 0.999953         | 1.000000        |

### Correlation between Latitude/Longitude and Performance

|                    | Latitude    | Longitude   | Subscribers  | Video Views  | Monthly Earnings | Yearly Earnings |
|--------------------|-------------|-------------|--------------|--------------|------------------|-----------------|
| Latitude           | 1.000000    | -0.200219   | 0.019450     | 0.037334     | 0.006873         | 0.006532        |
| Longitude          | -0.200219   | 1.000000    | 0.022443     | 0.031268     | 0.100299         | 0.100376        |
| Subscribers        | 0.019450    | 0.022443    | 1.000000     | 0.750958     | 0.388579         | 0.388935        |
| Video Views        | 0.037334    | 0.031268    | 0.750958     | 1.000000     | 0.551455         | 0.552091        |
| Monthly Earnings   | 0.006873    | 0.100299    | 0.388579     | 0.551455     | 1.000000         | 0.999953        |
| Yearly Earnings    | 0.006532    | 0.100376    | 0.388935     | 0.552091     | 0.999953         | 1.000000        |

### Correlation between Subscribers and Video Views

The correlation between subscribers and video views is strong at 0.75. This indicates a positive relationship between the number of subscribers a channel has and the number of video views it receives. In other words, as the number of subscribers increases, the video views tend to increase as well. This suggests that channels with a larger subscriber base are likely to have more views on their videos.

In [12]:
# 1. Top Countries by Number of Channels
top_countries_by_channels = youtube_data['Country'].value_counts().head(10)

# 2. Performance by Channel Type
performance_by_channel_type = youtube_data.groupby('channel_type').agg({'subscribers': 'mean', 'video views': 'mean', 'highest_monthly_earnings': 'mean', 'highest_yearly_earnings': 'mean'})

# 3. Top Channels by Subscriber Growth
top_channels_by_subscriber_growth = youtube_data.nlargest(10, 'subscribers_for_last_30_days')[['Youtuber', 'subscribers_for_last_30_days']]

# 4. Top Channels by Video Views Growth
top_channels_by_views_growth = youtube_data.nlargest(10, 'video_views_for_the_last_30_days')[['Youtuber', 'video_views_for_the_last_30_days']]

top_countries_by_channels, performance_by_channel_type, top_channels_by_subscriber_growth, top_channels_by_views_growth

(United States     313
 India             168
 Brazil             62
 United Kingdom     43
 Mexico             33
 Indonesia          28
 Spain              22
 Thailand           18
 South Korea        17
 Russia             16
 Name: Country, dtype: int64,
                 subscribers   video views  highest_monthly_earnings  \
 channel_type                                                          
 Animals        1.953333e+07  1.471175e+10              2.833333e+06   
 Autos          2.146667e+07  6.525849e+09              4.666701e+05   
 Comedy         2.085686e+07  8.885109e+09              7.444748e+05   
 Education      2.653469e+07  1.560828e+10              7.542673e+05   
 Entertainment  2.277138e+07  1.109175e+10              7.027955e+05   
 Film           2.296190e+07  1.093435e+10              4.646072e+05   
 Games          2.154694e+07  7.414359e+09              3.522757e+05   
 Howto          1.803056e+07  5.434740e+09              2.356722e+05   
 Music          2.67

## Top 10 Countries with the Most Channels

| Country         | Number of Channels |
|-----------------|-------------------|
| United States   | 313               |
| India           | 168               |
| Brazil          | 62                |
| United Kingdom  | 43                |
| Mexico          | 33                |
| Indonesia       | 28                |
| Spain           | 22                |
| Thailand        | 18                |
| South Korea     | 17                |
| Russia          | 16                |

## Channel Types and Performance

| Channel Type   | Subscribers       | Video Views      | Monthly Earnings  | Yearly Earnings   |
|----------------|-------------------|------------------|-------------------|-------------------|
| Animals        | 19,533,333        | 14,711,750,000   | $2,833,333        | $33,900,000       |
| Autos          | 21,466,667        | 6,525,849,000    | $466,670          | $5,666,707        |
| Comedy         | 20,856,864        | 8,885,109,000    | $744,475          | $8,963,068        |
| Education      | 26,534,687        | 15,608,280,000   | $754,267          | $9,034,639        |
| Entertainment  | 22,771,384        | 11,091,750,000   | $702,795          | $8,443,108        |
| Film           | 22,961,905        | 10,934,350,000   | $464,607          | $5,582,405        |
| Games          | 21,546,944        | 7,414,359,000    | $352,276          | $4,207,925        |
| Howto          | 18,030,559        | 5,434,740,000    | $235,672          | $2,840,464        |
| Music          | 26,719,906        | 15,058,970,000   | $571,582          | $6,875,110        |
| News           | 20,390,000        | 10,378,430,000   | $701,640          | $8,393,373        |
| Nonprofit      | 27,750,000        | 5,431,456,000    | $390,400          | $4,700,000        |
| People         | 19,339,601        | 8,771,476,000    | $675,705          | $8,106,875        |
| Sports         | 27,807,692        | 13,931,150,000   | $659,715          | $7,866,638        |
| Tech           | 18,870,586        | 3,491,502,000    | $220,359          | $2,635,200        |

## Top 10 Channels with the Most Subscribers in the Last 30 Days

| Youtuber        | Subscribers in the Last 30 Days |
|-----------------|---------------------------------|
| MrBeast         | 8,000,000                       |
| DaFuq!?Boom!    | 6,700,000                       |
| Jess No Limit   | 5,500,000                       |
| ZAMZAM ELECTRONICS TRADING | 3,400,000               |
| BeatboxJCOP     | 3,400,000                       |
| KL BRO Biju Rithvik | 3,200,000                    |
| ýýýýýýýýý KIMPRO | 3,200,000                       |
| Panda Shorts    | 3,000,000                       |
| BETER B��   | 2,100,000                       |
| T-Series        | 2,000,000                       |

## Top 10 Channels with the Most Video Views in the Last 30 Days

| Youtuber               | Video Views in the Last 30 Days |
|------------------------|---------------------------------|
| Happy Lives            | 6,589,000,000                   |
| Dan-Sa / Daniel Saboya | 6,148,000,000                   |
| ýýýýýýýýý KIMPRO       | 3,404,000,000                   |
| DaFuq!?Boom!           | 2,304,000,000                   |
| Calon Sarjana          | 2,292,000,000                   |
| T-Series               | 2,258,000,000                   |
| KL BRO Biju Rithvik    | 2,033,000,000                   |
| Cocomelon - Nursery Rhymes | 1,975,000,000               |
| SET India              | 1,824,000,000                   |
| Zee TV                 | 1,707,000,000                   |

In [13]:
# Correlation between Tertiary Education Enrollment and Performance
education_correlation = youtube_data[['Gross tertiary education enrollment (%)', 'subscribers', 'video views', 'highest_monthly_earnings', 'highest_yearly_earnings']].corr()

education_correlation

Gross tertiary education enrollment (%)  \
Gross tertiary education enrollment (%)                                 1.000000   
subscribers                                                            -0.006804   
video views                                                            -0.015232   
highest_monthly_earnings                                               -0.061973   
highest_yearly_earnings                                                -0.062163   

                                         subscribers  video views  \
Gross tertiary education enrollment (%)    -0.006804    -0.015232   
subscribers                                 1.000000     0.750958   
video views                                 0.750958     1.000000   
highest_monthly_earnings                    0.388579     0.551455   
highest_yearly_earnings                     0.388935     0.552091   

                                         highest_monthly_earnings  \
Gross tertiary education enrollment (%)                 -0.061973   
subscribers                                              0.388579   
video views                                              0.551455   
highest_monthly_earnings                                 1.000000   
highest_yearly_earnings                                  0.999953   

                                         highest_yearly_earnings  
Gross tertiary education enrollment (%)                -0.062163  
subscribers                                             0.388935  
video views                                             0.552091  
highest_monthly_earnings                                0.999953  
highest_yearly_earnings                                 1.000000

In [14]:
# Top Channels by Video Views Rank
top_channels_by_views_rank = youtube_data.nsmallest(10, 'video_views_rank')[['Youtuber', 'video_views_rank', 'video views']]

top_channels_by_views_rank

Youtuber  video_views_rank   video views
0                     T-Series               1.0  2.280000e+11
3   Cocomelon - Nursery Rhymes               2.0  1.640000e+11
4                    SET India               3.0  1.480000e+11
15                    Sony SAB               4.0  1.010000e+11
6          ýýý Kids Diana Show               5.0  9.324704e+10
11                         WWE               7.0  7.742847e+10
9                Vlad and Niki               8.0  7.718017e+10
21                      Zee TV               9.0  7.313905e+10
26                   Colors TV              10.0  6.151091e+10
29                  Movieclips              11.0  5.931647e+10

# **Please Upvote and comments for any suggestion and improvement**